deepspeed/model_implementations/transformers/ds_transformer.py

```diff
@@ -88,22 +84,27 @@ def __init__(self,
             self.norm_b = nn.Parameter(torch.empty(self.config.hidden_size, dtype=data_type, device=device),
                                        requires_grad=False)
         self.layer_past = None
-        try:
-            if config.dtype == torch.float32:
-                self.allocate_workspace = inference_module.allocate_workspace_fp32
-            elif config.dtype == torch.bfloat16:
-                self.allocate_workspace = inference_module.allocate_workspace_bf16
-            else:
-                self.allocate_workspace = inference_module.allocate_workspace_fp32
-            self._alloc_workspace = True
-        except AttributeError:
-            self.allocate_workspace = None
-            self._alloc_workspace = False
+        self.layer_norm = LayerNormOp()
+        DeepSpeedTransformerInference.workspace = WorkspaceOp(self.config)
```

Comment:
Should this not only happen if `DeepSpeedTransformerInference.layer_id == 1`?

Reply:
This logic is inside allocate_workspace function, here I just initialize the workspace op.